### PR TITLE
DCS-2026 decorate post arrival with body scan info

### PR DESCRIPTION
**Pull request by louisenorris**

### CODE REVIEW

File: server/data/__testutils/testObjects.ts [hunk 1]
```diff
@@ -18,14 +19,26 @@ import type { NewArrival } from '../../routes/bookedtoday/arrivals/state'
 import type { User } from '../hmppsAuthClient'
 import { MatchType } from '../../services/matchTypeDecorator'
 
-export const withBodyScanInfo = <T>(
+export const withBodyScanStatus = <T>(
```
Comment (louisenorris) [inline server/data/__testutils/testObjects.ts:22]: renamed specific to addition of body scan status only

File: server/services/bodyScanInfoDecorator.ts [hunk 2]
```diff
@@ -4,19 +4,32 @@ import { associateBy } from '../utils/utils'
...
   ) {}
 
-  public async decorate<T extends HasPrisonNumber>(items: T[]): Promise<WithBodyScanInfo<T>[]> {
+  public async decorate<T extends HasPrisonNumber>(items: T[]): Promise<WithBodyScanStatus<T>[]> {
     const token = await this.hmppsAuthClient.getSystemClientToken()
     const prisonNumbers = items.map(i => i.prisonNumber).filter(Boolean)
     const scanInfo = await this.bodyScanClientFactory(token).getBodyScanInfo(prisonNumbers)
     const prisonNumberToScan = associateBy(scanInfo, info => info.prisonNumber)
     return items.map(i => ({ ...i, bodyScanStatus: prisonNumberToScan.get(i.prisonNumber)?.bodyScanStatus }))
   }
```
Comment (louisenorris) [inline server/services/bodyScanInfoDecorator.ts:26]: Thought it best to keep current body scan decorator as is to update with `bodyScanStatus` only and not introduce additional body scan data to the recent arrival list page or elsewhere where we only need to know if the arrival can be scanned or not by referencing their `bodyScanStatus`

File: server/services/bodyScanInfoDecorator.ts [hunk 3]
```diff
@@ -4,19 +4,32 @@ import { associateBy } from '../utils/utils'
...
+
+  public async decorateSingle<T extends HasPrisonNumber>(item: T): Promise<WithBodyScanInfo<T>> {
+    const token = await this.hmppsAuthClient.getSystemClientToken()
+    const { numberOfBodyScans, numberOfBodyScansRemaining } = await this.bodyScanClientFactory(
+      token
+    ).getSingleBodyScanInfo(item.prisonNumber)
+    return { ...item, numberOfBodyScans, numberOfBodyScansRemaining }
+  }
```
Comment (louisenorris) [inline server/services/bodyScanInfoDecorator.ts:34]: Added a separate decorator to add body scan info to a single arrival.

File: server/@types/welcome/index.d.ts [hunk 4]
```diff
@@ -164,5 +165,16 @@ declare module 'welcome' {
...
+      firstName: string
+      lastName: string
+      dateOfBirth: string
+      prisonNumber: string
+      pncNumber?: string
+      croNumber?: string
+      sex: SexKeys
+      arrivalType: ArrivalType
+      arrivalTypeDescription: string
+    }
```
[the remark this answers is not in the document]
Comment (andrewrlee) [reply server/@types/welcome/index.d.ts:178]: think this stores types from the API.

I'd have thought prisoner details prison number shouldn't be nullable anyway?

[the remark this answers is not in the document]
Comment (andrewrlee) [reply server/@types/welcome/index.d.ts:178]: https://welcome-api-dev.prison.service.justice.gov.uk/swagger-ui/index.html#/prison-resource/getPrisoner

So think you can make it mandatory

[the remark this answers is not in the document]
Comment (louisenorris) [reply server/@types/welcome/index.d.ts:178]: I did try making it not optional but every multiple matches controller uses `getPrisonerDetails` and therefore the `PrisonerDetails` return type to select on of the multiple `PotentialMatch` which has an optional `prisonNumber`

[the remark this answers is not in the document]
Comment (louisenorris) [reply server/@types/welcome/index.d.ts:178]: Agree that it should be mandatory reflecting the response schema in the swagger docs but then maybe the multiple match controllers need a little tweaking

[the remark this answers is not in the document]
Comment (louisenorris) [reply server/@types/welcome/index.d.ts:178]: Looks like `prisonNumber` isn't optional in the response schema for potential matches in the swagger docs

[the remark this answers is not in the document]
Comment (louisenorris) [reply server/@types/welcome/index.d.ts:178]: Think I can fix typing issues by just casting the mock resolved type for `getPrisonerDetails` in the multiple match controller tests as `PrisonerDetails`

[the remark this answers is not in the document]
Comment (louisenorris) [reply server/@types/welcome/index.d.ts]: `prisonNumber` in `PotentialMatch` type made required to fix this issue and reflect backend in latest commit

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**